plugin/evm/validators/state_test.go

```diff
@@ -0,0 +1,288 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package validators
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+)
+
+func TestState(t *testing.T) {
+	require := require.New(t)
+	db := memdb.New()
+	state, err := NewState(db)
+	require.NoError(err)
+
+	// get non-existent uptime
+	nodeID := ids.GenerateTestNodeID()
+	vID := ids.GenerateTestID()
+	_, _, err = state.GetUptime(nodeID)
+	require.ErrorIs(err, database.ErrNotFound)
+
+	// set non-existent uptime
+	startTime := time.Now()
+	err = state.SetUptime(nodeID, 1, startTime)
+	require.ErrorIs(err, database.ErrNotFound)
+
+	// add new validator
+	state.AddNewValidator(vID, nodeID, uint64(startTime.Unix()), true)
+
+	// adding the same validator should fail
+	err = state.AddNewValidator(vID, nodeID, uint64(startTime.Unix()), true)
+	require.Error(err)
```

Comment:
should we add an error like ValidatorAlreadyExists or ValidatorAlreadyAdded and check it here?
also should we test this with a different nodeID (since the repeated vID will cause the failure)

Reply:
done